packages/browser/src/transports/new-xhr.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  BaseTransportOptions,
+  createTransport,
+  NewTransport,
+  TransportMakeRequestResponse,
+  TransportRequest,
+} from '@sentry/core';
+import { SyncPromise } from '@sentry/utils';
+
+/**
+ * The DONE ready state for XmlHttpRequest
+ *
+ * Defining it here as a constant b/c XMLHttpRequest.DONE is not always defined
+ * (e.g. during testing, it is `undefined`)
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState}
+ */
+const XHR_READYSTATE_DONE = 4;
+
+export interface XHRTransportOptions extends BaseTransportOptions {
+  headers?: { [key: string]: string };
```

Comment:
Decided to go with the `{[key: string]: string}` instead of `Record` as it is used that way in `TransportOptions`

https://github.com/getsentry/sentry-javascript/blob/caba96ef557464283b983466e5e490f3545d8c41/packages/types/src/transport.ts#L56

which are passed to `makeNewXHRTransport`. Seems to be more consistent that way IMHO